core/src/main/java/org/fao/geonet/api/records/attachments/AbstractStore.java

```diff
@@ -232,6 +232,13 @@ public final MetadataResource putResource(ServiceContext context, String metadat
         if (contentDisposition != null) {
             filename = ContentDisposition.parse(contentDisposition).getFilename();
         }
+        // If follow redirect, get the filename from the redirected URL
+        if (filename == null && connection.getInstanceFollowRedirects()) {
+            String redirectUrl = connection.getURL().toString();
+            if (redirectUrl != null) {
+                filename = getFilenameFromUrl(new URL(redirectUrl));
+            }
+        }
```

Comment:
`new URL(String)` is deprecated in Java 20. You can use `URI.toURL()` instead to avoid to use the deprecated method in case we upgrade to Java >= 20. This approach works seamlessly in Java 11 and avoids reliance on deprecated methods in future Java versions.

```suggestion
        // If follow redirect, get the filename from the redirected URL
        if (filename == null && connection.getInstanceFollowRedirects()) {
            String redirectUrl = connection.getURL().toString();
            if (redirectUrl != null) {
                URI redirectUri = URI.create(redirectUrl); // Create a URI object
                filename = getFilenameFromUrl(redirectUri.toURL()); // Convert URI to URL
            }
        }
```